File: app/src/main/java/com/wealoha/social/beans/HashTag.java
package com.wealoha.social.beans;

import java.io.Serializable;

public class HashTag implements Serializable {

	/**
	 *
	 */
	private static final long serialVersionUID = 8090400637875293423L;
	private String itemId;
	private String name;
	private CommonImage commonImage;
	private String summary;
	private int postCount;
	private String url;
	private String type;
	private int userCount;

	public HashTag(String itemId, String name, CommonImage commonImage, String summary, int postCount, String url, String type, int userCount) {
		super();
		this.itemId = itemId;
		this.name = name;
		this.commonImage = commonImage;
		this.summary = summary;
		this.postCount = postCount;
		this.url = url;
		this.type = type;
		this.userCount = userCount;
	}

	public static HashTag fromDTO(HashTagDTO dto) {
		if (dto == null) {
			return null;
		}
		return new HashTag(dto.itemId, dto.name, CommonImage.fromDTO(dto.backgroundImage), dto.summary, dto.postCount, dto.url, dto.type, dto.userCount);
	}

	public String getItemId() {
		return itemId;
	}

	public String getName() {
		return name;
	}

	public CommonImage getBackgroundImage() {
		return commonImage;
	}

	public String getSummary() {
		return summary;
	}

	public int getPostCount() {
		return postCount;
	}

	public String getUrl() {
		return url;
	}

	public String getType() {
		return type;
	}

	public int getUserCount() {
		return userCount;
	}

}

File: app/src/main/java/com/wealoha/social/beans/HashTag.kt
package com.wealoha.social.beans

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import java.io.Serializable

@Parcelize
data class HashTag(
    val itemId: String?,
    val name: String?,
    val backgroundImage: CommonImage?,
    val summary: String?,
    val postCount: Int,
    val url: String?,
    val type: String?,
    val userCount: Int
) : Serializable, Parcelable {

    companion object {
        /**
         *
         */
        private const val serialVersionUID = 8090400637875293423L
        @kotlin.jvm.JvmStatic
        fun fromDTO(dto: HashTagDTO?): HashTag? {
            return if (dto == null) {
                null
            } else HashTag(
                dto.itemId,
                dto.name,
                CommonImage.fromDTO(dto.backgroundImage!!),
                dto.summary,
                dto.postCount,
                dto.url,
                dto.type,
                dto.userCount
            )
        }
    }

}